diff -uNp openssl-1.0.0d/crypto/ecdsa/ecs_ossl.c openssl-1.0.0d-with-ec-patch/crypto/ecdsa/ecs_ossl.c
--- openssl-1.0.0d/crypto/ecdsa/ecs_ossl.c	2009-12-01 18:32:33.000000000 +0100
+++ openssl-1.0.0d-with-ec-patch/crypto/ecdsa/ecs_ossl.c	2011-05-23 16:53:18.745999999 +0200
@@ -144,6 +144,16 @@ static int ecdsa_sign_setup(EC_KEY *ecke
 			}
 		while (BN_is_zero(k));
 
+		#ifdef ECDSA_POINT_MUL_NO_CONSTTIME
+		 /* We do not want timing information to leak the length of k,
+		  * so we compute G*k using an equivalent scalar of fixed
+		  * bit-length. */
+
+		 if (!BN_add(k, k, order)) goto err;
+		 if (BN_num_bits(k) <= BN_num_bits(order))
+		 if (!BN_add(k, k, order)) goto err;
+		#endif /* def(ECDSA_POINT_MUL_NO_CONSTTIME) */
+
 		/* compute r the x-coordinate of generator * k */
 		if (!EC_POINT_mul(group, tmp_point, k, NULL, NULL, ctx))
 		{
diff -uNp openssl-1.0.0d/crypto/ecdsa/Makefile openssl-1.0.0d-with-ec-patch/crypto/ecdsa/Makefile
--- openssl-1.0.0d/crypto/ecdsa/Makefile	2009-10-18 16:44:50.000000000 +0200
+++ openssl-1.0.0d-with-ec-patch/crypto/ecdsa/Makefile	2011-05-23 17:04:38.004000728 +0200
@@ -10,7 +10,7 @@ CFLAG=-g -Wall
 MAKEFILE=	Makefile
 AR=		ar r
 
-CFLAGS= $(INCLUDES) $(CFLAG)
+CFLAGS= $(INCLUDES) $(CFLAG) -DECDSA_POINT_MUL_NO_CONSTTIME
 
 GENERAL=Makefile
 TEST=ecdsatest.c
